src/server/response.cpp

```diff
@@ -93,15 +93,16 @@ std::unique_ptr<Response> Response::build_304(const InternalServer& server, cons
   return response;
 }
 
-std::unique_ptr<Response> Response::build_404(const InternalServer& server, const RequestContext& request, const std::string& bookName)
+std::unique_ptr<Response> Response::build_404(const InternalServer& server, const RequestContext& request, const std::string& bookName, const std::string& details)
 {
   MustacheData results;
   results.set("url", request.get_full_url());
-
+  results.set("details", details);
+  results.set("bookName", bookName);
   auto response = ContentResponse::build(server, RESOURCE::templates::_404_html, results, "text/html");
   response->set_code(MHD_HTTP_NOT_FOUND);
   response->set_taskbar(bookName, "");
-
+  response->set_search_link(bookName, details);
   return std::move(response);
 }
```

Comment:
Render the details of a 404 response outside this function and pass `details` as a fragment of HTML that can be embedded in the response as is.

Reply:
@veloman-yunkan I did this in a different way, but I think it's similar to what you advised